Documentation/ceph-object-bucket-claim.md

```diff
@@ -0,0 +1,242 @@
+
+## Sample
+
+##### OBC Custom Resource (User Defined)
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: ceph-bucket [1]
+  namespace: rook-ceph [2]
+spec:
+  bucketName: [3]
+  generateBucketName: photo-booth [4]
+  storageClassName: rook-ceph-bucket [5]
+  additionalConfig: [6]
+    ANY_KEY: VALUE ...
+```
+1. `name` of the `ObjectBucketClaim`. This name becomes the name of the Secret and ConfigMap.
+1. `namespace`(optional) of the `ObjectBucketClaim`, which is also the namespace of the ConfigMap and Secret. If nothing is specified it uses `default`
+1. `name` of the `bucket`. If supplied then `generateBucketName` is ignored.
+**Not** recommended for new buckets since names must be unique within
+an entire object store.
+1. if supplied then `bucketName` must be empty. This value becomes the prefix for a randomly generated name.
+If both `bucketName` and `generateBucketName` are supplied then `BucketName` has precedence and `GenerateBucketName` is ignored.
+If both `bucketName` and `generateBucketName` are blank or omitted then the storage class is expected to contain the name of an _existing_ bucket. It's an error if all three bucket related names are blank or omitted.
+1. `storageClassName` which defines the object-store service and the bucket provisioner.
+1. `additionalConfig` gives providers a location to set proprietary config values (tenant, namespace...).
+The value is a list of 1 or more key-value pairs.
+
+##### OBC Custom Resource (after update by lib)
+```yaml
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  finalizers: [1]
+  - objectbucket.io/finalizer
+  labels: [2]
+    bucket-provisioner: aws-s3.io-bucket [3]
+spec:
+  bucketName: photo-booth-62PrQ [4]
+  objectBucketRef: objectReference{} [5]
+  configMapRef: objectReference{} [6]
+  secretRef: objectReference{} [7]
+status:
+  phase: {"Pending", "Bound", "Released", "Failed"} [8]
+```
+1. the `finalizer` added by the library, the name is a constant.
+1. the library adds a `label` (seen here) but each provisioner can
+   supply their own `labels`.
+1. the label value shown is the name of the provisioner.
+   In this example the provisioner name is `aws-s3.io/bucket`.
+1. the generated, unique `bucket name` for the new bucket.
+1. `objectReference` to the generated `OB`.
+1. `objectReference` to the generated `ConfigMap`.
+1. `objectReference` to the generated `Secret`.
+1. phases of bucket creation:
+    - _Pending_: the operator is processing the request
+    - _Bound_: the operator finished processing the request and linked the OBC and OB
+    - _Released_: the OB has been deleted, leaving the OBC unclaimed but unavailable.
+    - _Failed_: not currently set.
+
+##### Generated Secret (sample for rook-ceph provider)
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ceph-bucket
```

Comment:
add number here so you can mention (again) that the secret name is the same as the OBC name.